### ISSUE TITLE
Update Readme :: Contribution Guidelines

### ISSUE THREAD

**Issue by arpansac** [commudle/commudle-ng#3]
Need to include:
- Branch naming conventions (feature branch, fix branch, documentation branch)
- All the PR's should be directed towards 'development' branch
- Commudle features + Screenshots (preferably gifs)
- Explanation of the folder structure

**Comment by IshetaBansal** [comment]
I would like to work on this issue. Please assign it to me.

**Comment by arpansac** [comment]
@IshetaBansal done, assigned!
Let's define a timeline as well please, approximately till when would you be able to complete it?

**Comment by smriti1313** [comment]
Hi, is there any chance of me contributing in this issue? I would really like to work on it.

**Comment by arpansac** [comment]
@smriti1313 I think this is more of a 1 person task....
If you and @IshetaBansal can both divide the tasks, would be happy to co-assign this to you too!

**Comment by smriti1313** [comment]
@IshetaBansal Will it be alright with you to share this task with me? We can divide the work in this.

**Comment by IshetaBansal** [comment]
> @IshetaBansal Will it be alright with you to share this task with me? We can divide the work in this.

Hey!

**Comment by IshetaBansal** [comment]
> @IshetaBansal done, assigned!
> Let's define a timeline as well please, approximately till when would you be able to complete it?

I'll try to complete it by tomorrow evening.

**Comment by IshetaBansal** [comment]
I

> Need to include:
> 
> * Branch naming conventions (feature branch, fix branch, documentation branch)
> * All the PR's should be directed towards 'development' branch
> * Commudle features + Screenshots (preferably gifs)
> * Explanation of the folder structure

Hey
Could elaborate on the branch naming conventions? I am unable to find any of these except the development branch.

**Comment by arpansac** [comment]
> I
> 
> > Need to include:
> > 
> > * Branch naming conventions (feature branch, fix branch, documentation branch)
> > * All the PR's should be directed towards 'development' branch
> > * Commudle features + Screenshots (preferably gifs)
> > * Explanation of the folder structure
> 
> Hey! Following are some things I want to confirm for working on this issue:
> 
> > Could you elaborate on the branch naming conventions? I am unable to find any of these except the development branch.
> > And where can I find the features of Commudle in brief?
> > What do you mean by the folder structure?
> > Waiting for your answer :)

Hey! Let me try to answer inline

Could you elaborate on the branch naming conventions? I am unable to find any of these except the development branch.
- So these are usually done in a way `feature-<branchname>`, `fix-<branchname>` and so on. Try to google a little and you might find something even better (do post it here if it's interesting, before making those changes)

And where can I find the features of Commudle in brief?
- Majorly there's 6 things: Community groups, Communities, Events, Community Channels, Labs and Builds. Please send an email to arpan@commudle.com and I'll add you to one of the demo communities as an organizer to explore more on the admin side.
What do you mean by the folder structure?
- When someone starts to code, what types of files go where. For this you'll need to got through the repository, exploring what goes where. 

Relax, don't be in a hurry, this will actually take a little bit of Googling and setting up the project on your system too. It will give a good insight on the best practices for scalable projects.

**Comment by arpansac** [comment]
> @IshetaBansal Will it be alright with you to share this task with me? We can divide the work in this.

@smriti1313 will you be interested to take this up? Unfortunately we didn't get any updates from the previous assignee

**Comment by smriti1313** [comment]
> > @IshetaBansal Will it be alright with you to share this task with me? We can divide the work in this.
> 
> @smriti1313 will you be interested to take this up? Unfortunately we didn't get any updates from the previous assignee

@arpansac Thank you so much but i am sorry since i got my end sem exams for now. You may assign it to the other person please. Thank you.

**Comment by IshetaBansal** [comment]
> > @IshetaBansal Will it be alright with you to share this task with me? We can divide the work in this.
> 
> @smriti1313 will you be interested to take this up? Unfortunately we didn't get any updates from the previous assignee

Hey
I'd been working previously on this issue and I've been MIA due to some family emergency. I would be highly obliged if you reassigned the issue to me.
Thank you :)

**Comment by arpansac** [comment]
> > > @IshetaBansal Will it be alright with you to share this task with me? We can divide the work in this.
> > 
> > 
> > @smriti1313 will you be interested to take this up? Unfortunately we didn't get any updates from the previous assignee
> 
> Hey
> I'd been working previously on this issue and I've been MIA due to some family emergency. I would be highly obliged if you reassigned the issue to me.
> Thank you :)


Done. Please take care.

**Comment by kanishkaa24** [comment]
I would love to work on this issue.

**Comment by arpansac** [comment]
> I would love to work on this issue.

@kanishkaa24 we're waiting for @IshetaBansal to make a progress on this. Please choose another issue :)

**Comment by arpansac** [comment]
@IshetaBansal please update.

**Comment by arpansac** [comment]
@kanishkaa24 would you still want to work on this? If yes, please list the changes we'd need to do in the readme. For that you will have to setup the code on your system and see where all does a developer get stuck.

**Comment by kanishkaa24** [comment]
Definitely. I'll add images and gifs. If you need me to add any more things, I'm willing to do that too.

**Comment by arpansac** [comment]
> Definitely. I'll add images and gifs. If you need me to add any more things, I'm willing to do that too.

Cool, have you setup the code on your system?

**Comment by kanishkaa24** [comment]
Do I need to setup the code?

**Comment by arpansac** [comment]
> Do I need to setup the code?

Yup, setting it up will help understand the process and improve the readme steps where it lists how to setup the code with some screenshots :)

**Comment by kanishkaa24** [comment]
Do you need me to add contribution guidelines too?
What is the procedure for setting up the code?
Same in the README.md?

**Comment by arpansac** [comment]
Contribution guidelines are already present, but if you think something
important can be improved/added, then sure!

Readme has it all.

**Comment by arpansac** [comment]
Hey @kanishkaa24 I've been super occupied with some launch. Will get on this tomorrow

**Comment by Dhriti-Roy** [comment]
Is this issue open. I would like to do it. kindly assign me for this issue.

**Comment by arpansac** [comment]
Finally, for more than a year now, we have a good readme, tried and tested. Thank you so much, everyone, for all the support.
